src/main/kotlin/no/nav/familie/ef/sak/selvstendig/NæringsinntektKontrollService.kt

```diff
@@ -35,67 +37,153 @@ class NæringsinntektKontrollService(
+                        oppgaveService.oppdaterOppgave(oppgaveMedUtsattFrist)
+                    } else {
+                        // Lag notat
+                        giBeskjedOmKontrollertInntektVedLøpendeOvergangsstønad(it.behandlingId, it.personIdent)
+                        val avsluttOppgaveMedOppdatertBeskrivelse = it.oppgave.copy(beskrivelse = it.oppgave.beskrivelse + "\nAutomatisk avsluttet oppgave: Ingen endring i inntekt.", status = StatusEnum.FERDIGSTILT)
+                        oppgaveService.oppdaterOppgave(avsluttOppgaveMedOppdatertBeskrivelse)
                     }
+                } else {
+                    beOmRegnskap(it.personIdent, it.behandlingId)
                 }
             }
+            opprettFremleggHvisOvergangsstønadMerEnn4MndIÅr(it.behandlingId, it.oppgave, it.tilkjentYtelse)
         }
+
         return fagsakIds
     }
 
+    private fun hentNæringsinntektDataForBeregning(oppgaver: List<Oppgave>): List<NæringsinntektDataForBeregning> {
+        val næringsinntektDataForBeregningList =
+            oppgaver.map {
+                val personIdent =
+                    it.identer?.firstOrNull { it.gruppe == IdentGruppe.FOLKEREGISTERIDENT }?.ident
+                        ?: it.personident ?: throw Exception("Fant ikke registrert ident på oppgave ${it.id}")
+                val fagsakOvergangsstønad = fagsakService.finnFagsaker(setOf(personIdentForOppgave(it))).firstOrNull { it.stønadstype == StønadType.OVERGANGSSTØNAD } ?: throw RuntimeException("Fant ikke fagsak for overgangsstønad for person: ${personIdentForOppgave(it)}")
+                val behandlingId = behandlingService.finnSisteIverksatteBehandling(fagsakOvergangsstønad.id)?.id ?: throw RuntimeException("Fant ingen gjeldende behandling for fagsakId: ${fagsakOvergangsstønad.id}")
+                val tilkjentYtelse = tilkjentYtelseService.hentForBehandling(behandlingId)
+                val næringsinntektDataForBeregning =
+                    NæringsinntektDataForBeregning(
+                        oppgave = it,
+                        personIdent = personIdent,
+                        fagsak = fagsakOvergangsstønad,
+                        behandlingId = behandlingId,
+                        tilkjentYtelse = tilkjentYtelse,
+                        fjoråretsNæringsinntekt = hentFjoråretsNæringsinntekt(fagsakOvergangsstønad.fagsakPersonId),
+                        fjoråretsPersonInntekt = inntektService.hentÅrsinntekt(personIdent, årstallIFjor),
+                        forventetInntektIFjor = forventetInntektSnittIFjor(tilkjentYtelse),
+                    )
+                secureLogger.info("Næringsinntektsdata for beregning: $næringsinntektDataForBeregning")
+                næringsinntektDataForBeregning
+            }
+
+        return næringsinntektDataForBeregningList
+    }
+
+    private fun personIdentForOppgave(it: Oppgave) =
```

Comment:
Det er en del logikk i denne servicen som kan være litt krevende å få hodet igjennom. Navngivningen er generelt god, og mange fine utils-funksjoner med gode navn. 

Hadde det vært konseptuelt lettere å forstå denne servicen hvis utils-funksjonene lå i en NæringsinntektKontrollUtils.kt? 

Tenker da på `personIdentForOppgave`, `oppfyllerAktivitetsplikt`, `skalKontrolleres`, `forventetInntektSnittIFjor`, `antallMånederMedVedtakForÅr` 


Eller blir det bare mer rot å ha ting på to plasser?

Reply:
Enig! Var litt for sparsom på egne klasser/filer her ser jeg. Endte med å lage en egen `NæringsinntektDataForBeregningService` hvor all innhenting og mapping av data skjer. I tillegg lagde jeg egen fil for klassen `NæringsinntektDataForBeregning` hvor jeg kunne legge inn noen metoder som ikke trenger andre avhengigheter, som f.eks. `skalKontrolleres()` og `oppfyllerAktivitetsplikt()`.